src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,207 @@
+                         = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            return statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't delete a car by id " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String getCarsForDriverRequest = "SELECT car_id, model, manufacturer_id, name, country "
+                + "FROM cars AS c" + "    INNER JOIN cars_drivers cd ON c.id = cd.car_id"
+                + "    INNER JOIN manufacturers m ON c.manufacturer_id  = m.id"
+                + "    WHERE c.is_deleted = FALSE AND driver_id = ?";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement =
+                         connection.prepareStatement(getCarsForDriverRequest)) {
+            statement.setLong(1, driverId);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(getCar(resultSet));
+            }
+            return cars;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get a list of cars from carsDB.", e);
+        }
+    }
+
+    private Car getCar(ResultSet resultSet) throws SQLException {
+        Long id = resultSet.getObject("car_id", Long.class);
+        String model = resultSet.getString("model");
+        Long manufacturerId = resultSet.getObject("manufacturer_id", Long.class);
+        String manufacturerName = resultSet.getString("name");
+        String manufacturerCountry = resultSet.getString("country");
+        Manufacturer manufacturer =
+                new Manufacturer(manufacturerId, manufacturerName, manufacturerCountry);
+        Car car = new Car();
+        car.setId(id);
+        car.setModel(model);
+        car.setManufacturer(manufacturer);
+        car.setDrivers(getDriversForCar(id));
+        insertDrivers(car);
+        return car;
+    }
+
+    private List<Driver> getDriversForCar(Long carId) {
+        String getAllDriversForCarRequest = "SELECT id, name, license_number "
+                + "FROM drivers d JOIN cars_drivers cd "
+                + "ON d.id = cd.driver_id WHERE cd.car_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement getAllDriversStatement =
+                         connection.prepareStatement(getAllDriversForCarRequest)) {
+            getAllDriversStatement.setLong(1, carId);
+            ResultSet resultSet = getAllDriversStatement.executeQuery();
+            List<Driver> drivers = new ArrayList<>();
+            while (resultSet.next()) {
+                Driver driver = parseDriversFromResultSet(resultSet);
+                if (drivers.contains(driver)) {
+                    continue;
+                }
```

Comment:
you dont need this check - you should avoid having duplicates in cars_drivers table